src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -33,6 +37,12 @@ public LogicManager(Model model, Storage storage) {
     public CommandResult execute(String commandText) {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
         Command command = parser.parseCommand(commandText);
+        
+        if(command.modifiesData()){
+        	model.recordStateBeforeChange(command);
+        	model.resetRedoables();
+        }
+        
```

Comment:
So, if you store the state *before* the command is run, you run into all kinds of trouble like not knowing if the command actually modifies data or not (which is why you needed the `command.modifiesData()`), or if the command actually ran undo/redo or not.

If you do the check *after* the command is run, however...